internal/harnesses/remote/k8s.go

```diff
@@ -0,0 +1,382 @@
+package remote
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/url"
+	"slices"
+
+	"github.com/chainguard-dev/terraform-provider-imagetest/internal/harnesses/base"
+	"github.com/chainguard-dev/terraform-provider-imagetest/internal/types"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/httpstream"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+const (
+	ImagetestNamespaceNameFormat  = "imagetest-%s"
+	ImagetestSandboxPodNameFormat = "%s-sandbox"
+	EntrypointContainerName       = "wolfi-base"
+	RbacResourcesNameFormat       = "rbac-%s"
+)
+
+// Ensure type k8s conforms to types.Harness.
+var _ types.Harness = &k8s{}
+
+type k8s struct {
+	// Provide basic functions needed to make the harness operate.
+	*base.Base
+
+	// ID to identify resources created by this harness.
+	Id string
+
+	// RESTClient config.
+	Config *rest.Config
+
+	// Kubernetes Client instance.
+	Client kubernetes.Interface
+}
+
+func New(id string, kubeconfig *string) (types.Harness, error) {
+	rules := clientcmd.NewDefaultClientConfigLoadingRules()
+	rules.DefaultClientConfig = &clientcmd.DefaultClientConfig
+
+	if kubeconfig != nil {
+		rules.ExplicitPath = *kubeconfig
+	}
+
+	overrides := &clientcmd.ConfigOverrides{ClusterDefaults: clientcmd.ClusterDefaults}
+	config, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(rules, overrides).ClientConfig()
+	if err != nil {
+		return nil, fmt.Errorf("failed to create configuration: %w", err)
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Kubernetes client: %w", err)
+	}
+
+	return &k8s{
+		Base:   base.New(),
+		Id:     id,
+		Config: config,
+		Client: clientset,
+	}, nil
+}
+
+func (h *k8s) Setup() types.StepFn {
+	return h.WithCreate(func(ctx context.Context) (context.Context, error) {
+		var rootUserID int64 = 0
+		commonLabels := map[string]string{
+			"dev.chainguard.imagetest":                    "true",
+			"dev.chainguard.imagetest/parent-harness-ref": h.Id,
+		}
+
+		namespaceName := fmt.Sprintf(ImagetestNamespaceNameFormat, h.Id)
+		sandboxPodName := fmt.Sprintf(ImagetestSandboxPodNameFormat, h.Id)
+		rbacResourcesName := fmt.Sprintf(RbacResourcesNameFormat, h.Id)
+
+		_, err := h.Client.CoreV1().Namespaces().Create(ctx, &corev1.Namespace{
```

Comment:
I think you want to handle (skip) the error where the namespace exists, ala `Apply()`

Reply:
Updated